clients/admin-ui/src/features/poc/privacy-notices-sandbox/PrivacyNoticeSandboxSimulated.tsx

```diff
@@ -0,0 +1,196 @@
+      setCheckedKeys(mechanismDefaults);
+      setSavedCheckedKeys(mechanismDefaults);
+      setPreviousSavedKeys(mechanismDefaults);
+      setDbState({});
+    },
+    [mechanismDefaults],
+  );
+
+  const handleSave = useCallback(() => {
+    setPreviousSavedKeys(savedCheckedKeys); // Store the previous saved state
+    setSavedCheckedKeys(checkedKeys); // Update saved state to current
+
+    // Update DB state with changes
+    const changedKeys = ALL_KEYS.filter((key) => {
+      const isCurrentlyChecked = checkedKeys.includes(key);
+      const wasPreviouslyChecked = savedCheckedKeys.includes(key);
+      return isCurrentlyChecked !== wasPreviouslyChecked;
+    });
+
+    setDbState((prev) => {
+      const newState = { ...prev };
+      changedKeys.forEach((key) => {
+        const noticeKey =
+          key === PARENT_KEY_WITH_UUID
+            ? PARENT_KEY
+            : TREE_NODES.find((node) => node.key === key)?.title ||
+              key.toString();
+        const isChecked = checkedKeys.includes(key);
+        newState[noticeKey] = isChecked ? "opt_in" : "opt_out";
+
+        // If cascade is enabled and parent was changed, also update all children
+        if (cascadeConsent && key === PARENT_KEY_WITH_UUID) {
+          const value = isChecked ? "opt_in" : "opt_out";
+          TREE_NODES.forEach((node) => {
+            newState[node.title] = value;
+          });
+        }
+      });
+      return newState;
+    });
+  }, [checkedKeys, savedCheckedKeys, cascadeConsent]);
+
+  return (
+    <div className="mt-5">
+      <Flex gap={24}>
+        {/* Left Column - Consent Management */}
+        <Flex vertical flex={1}>
+          <Typography.Text strong className="mb-4 block">
+            Notice configuration
+          </Typography.Text>
+          <ConsentMechanismRadioGroup
+            mechanism={consentMechanism}
+            onMechanismChange={handleMechanismChange}
+          />
+          <Typography.Text strong className="mb-4 block">
+            Consent behavior
+          </Typography.Text>
+          <CascadeConsentToggle
+            isEnabled={cascadeConsent}
+            onToggle={handleCascadeToggle}
+          />
+
+          <Typography.Text strong className="mb-4 block">
+            User preferences
+          </Typography.Text>
+          <PrivacyNoticesTree
+            privacyNotices={treeNotices}
+            checkedKeys={checkedKeys}
+            onCheckedKeysChange={setCheckedKeys}
+            cascadeConsent={cascadeConsent}
+          />
+
+          <Button type="primary" onClick={handleSave} className="mt-4">
+            Save preferences
+          </Button>
+        </Flex>
+
+        {/* Right Column - POST API Preview */}
+        <Flex vertical flex={1} style={{ minWidth: 0 }}>
+          <PostApiPreview
+            currentSavedKeys={toCheckedKeysType(savedCheckedKeys)}
+            previousSavedKeys={toCheckedKeysType(previousSavedKeys)}
```

Comment:
This adds code to the compiled JS unnecessarily, just for simple TS coersion.
```suggestion
            currentSavedKeys={savedCheckedKeys as CheckedKeysType}
            previousSavedKeys={previousSavedKeys as CheckedKeysType}
```
(can remove the util)
but even this is probably overkill, the type checker doesn't seem to have a problem with just doing
```suggestion
            currentSavedKeys={savedCheckedKeys}
            previousSavedKeys={previousSavedKeys}
```